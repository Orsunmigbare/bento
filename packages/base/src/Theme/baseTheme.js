import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
