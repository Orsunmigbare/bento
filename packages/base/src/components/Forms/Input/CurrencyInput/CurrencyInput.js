import React from "react"
import Input, { InputDefaultProps, InputPropTypes } from "../Input"
import {
  createCurrencyMask,
  currencyMaskOptions
} from "../../../../functions/Currency/currency"

const CurrencyInput = ({ ...props }) => {
  const createMask = createCurrencyMask({ ...props })
  Object.keys(currencyMaskOptions).forEach(key => {
    delete props[key]
  })
  delete props["mask"]
  const value = props.value ? props.value.toString() : '0'
  const mask = createMask(value)
  return <Input type="number" mask={mask} {...props} />
}

CurrencyInput.propTypes = InputPropTypes
CurrencyInput.defaultProps = {
  ...InputDefaultProps
}

export default CurrencyInput
