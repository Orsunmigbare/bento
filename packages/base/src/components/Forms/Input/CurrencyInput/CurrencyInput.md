```jsx harmony
<CurrencyInput value={12341234}/>
```

```jsx harmony
<CurrencyInput value={12341234} prefix={'kr '} suffix={',-'}/>
```

```jsx harmony
<CurrencyInput value={12341234} prefix={'kr '} suffix={',-'} thousandsSeparatorSymbol={'.'}/>
```

```jsx harmony
<CurrencyInput value={1234.01} requireDecimal={true}/>
```

```jsx harmony
<div>
    <CurrencyInput value={1234.01} suffix={' EUR'}/>
    <CurrencyInput value={1234.01} prefix={'â‚¬'} suffix={''}/>
</div>
```