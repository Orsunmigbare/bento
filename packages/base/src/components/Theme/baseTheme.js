export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
