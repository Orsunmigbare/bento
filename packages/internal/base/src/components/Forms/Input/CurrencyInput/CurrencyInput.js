import React from "react"
import PropTypes from "prop-types"
import Input, { InputPropTypes } from "../Input"
/**
 * Input for numbers and Currencies with possibilities to add prefix/suffix and optional thousand-seperators.
 */
const CurrencyInput = ({ prefix = null, delimiter = " ", ...props }) => {
  return (
    <Input
      type="text"
      options={{ numeral: true, delimiter, prefix }}
      pattern="\d*"
      {...props}
    />
  )
}

/* TODO: Should have a pattern so that we get number-keyboard: added this: https://stackoverflow.com/questions/6178556/phone-numeric-keyboard-for-text-input */

CurrencyInput.propTypes = {
  ...InputPropTypes,
  id: PropTypes.string.isRequired,
  suffix: PropTypes.string,
  prefix: PropTypes.string,
  thousandsSeparatorSymbol: PropTypes.string
}

CurrencyInput.defaultProps = {
  suffix: "",
  prefix: "",
  thousandsSeparatorSymbol: " "
}

export default CurrencyInput
