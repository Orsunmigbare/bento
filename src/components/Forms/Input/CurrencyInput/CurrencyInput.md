```jsx harmony
<CurrencyInput value={12341234} id="ieoh12ehiw" />
```

```jsx harmony
<CurrencyInput value={12341234} prefix={'kr '} suffix={',-'} id="rklfnj3" />
```

```jsx harmony
<CurrencyInput value={12341234} prefix={'kr '} suffix={',-'} thousandsSeparatorSymbol={'.'} id="wqkijdfhu" />
```

```jsx harmony
<div>
    <CurrencyInput value={1234}  suffix={' EUR'} id="23lknr"/>
    <CurrencyInput value={1234} prefix={'â‚¬'} suffix={''} id="oihfuwes"/>
</div>
```
