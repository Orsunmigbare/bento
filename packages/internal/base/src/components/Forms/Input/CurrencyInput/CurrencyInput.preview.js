import React from "react"
import CurrencyInput from "./CurrencyInput"
import Input from "../Input"
import Layout from "../../../Layout/Layout/Layout"

const preview = {
  title: "CurrencyInput",
  category: "component/Forms/Input",
  component: CurrencyInput,
  render: props => <CurrencyInput id="rklfnj3" {...props} />,
  renderExampleController: ({
    setComponentState,
    suffix,
    prefix,
    thousandsSeparatorSymbol
  }) => (
    <Layout grid="form">
      <Input
        label="Prefix"
        id="Prefixexample"
        defaultValue={prefix}
        onChange={e => setComponentState({ prefix: e.target.value })}
      />
      <Input
        label="Suffix"
        id="suffixexample"
        defaultValue={suffix}
        onChange={e => setComponentState({ suffix: e.target.value })}
      />
      <Input
        label="Thousands Separator Symbol"
        id="thousandsSeparatorSymbol"
        defaultValue={thousandsSeparatorSymbol}
        onChange={e =>
          setComponentState({ thousandsSeparatorSymbol: e.target.value })
        }
      />
    </Layout>
  )
}

export default preview
