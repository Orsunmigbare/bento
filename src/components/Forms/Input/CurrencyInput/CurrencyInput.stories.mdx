import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import CurrencyInput from "./CurrencyInput"

<Meta title="Components|Forms/Input/CurrencyInput" component={CurrencyInput}/>


# CurrencyInput
This components controlls user input so that they can only enter numbers

# import

```javascript
import {CurrencyInput} from "@staccx/bento"
```

# Types of CurrencyInputs


## Standard CurrencyInput
The inputfield only allowed numbers and formats the input with thousand seperators


<Preview>
  <Story name="Standard">
   <CurrencyInput label={"CurrencyInput"} />
  </Story>
</Preview>



# Props for CurrencyInput
<Props of={CurrencyInput} />
