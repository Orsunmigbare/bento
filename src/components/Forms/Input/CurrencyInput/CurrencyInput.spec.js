import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import CurrencyInput from "./CurrencyInput"

describe("Alert", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <CurrencyInput id="test" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
