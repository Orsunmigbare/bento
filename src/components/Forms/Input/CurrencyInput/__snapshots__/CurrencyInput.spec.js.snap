// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshots No props 1`] = `
<div
  className="sc-AykKI iDYrmt"
>
  
  
  <input
    autoFocus={null}
    className="sc-AykKJ gjiytr"
    disabled={false}
    hidden={false}
    id="test"
    name={null}
    onBlur={null}
    onChange={null}
    onKeyDown={null}
    placeholder={null}
    style={null}
    type="tel"
  />
</div>
`;
